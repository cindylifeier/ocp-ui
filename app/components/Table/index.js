/**
 *
 * Table
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const Table = styled.div`
  margin: 10px;
  background-color: white;
  min-width: 450px;
`;


Table.propTypes = {
  children: PropTypes.node,
};

export default Table;
