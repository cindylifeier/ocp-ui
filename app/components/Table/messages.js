/*
 * Table Messages
 *
 * This contains all the text for the Table component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'ocpui.components.Table.header',
    defaultMessage: 'This is the Table component !',
  },
});
