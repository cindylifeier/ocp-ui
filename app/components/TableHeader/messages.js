/*
 * TableHeader Messages
 *
 * This contains all the text for the TableHeader component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'ocpui.components.TableHeader.header',
    defaultMessage: 'This is the TableHeader component !',
  },
});
