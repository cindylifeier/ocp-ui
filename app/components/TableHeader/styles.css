.rowGridContainer:nth-child(odd) {
  background-color: #f2f2f2;
}

.rowGridContainer:first-child {
  background-color: #f2f2f2;
  font-size: 20px;
}

.rowGridContainer {
  display: grid;
  grid-gap: 5px;
  border-bottom: 1px outset rgb(51, 51, 51);
}
