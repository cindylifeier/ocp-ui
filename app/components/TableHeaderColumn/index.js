/**
 *
 * TableHeaderColumn
 *
 */


import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableHeaderColumn = styled.div`
  color: rgb(51, 51, 51);
  font-family: "Arial Bold", "Arial", sans-serif;
  font-style: normal;
  font-size: 15px;
  font-weight: 700;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-word;
  padding: 2px;
`;

TableHeaderColumn.propTypes = {
  children: PropTypes.node,
};

export default TableHeaderColumn;
