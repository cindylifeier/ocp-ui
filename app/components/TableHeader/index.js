/**
 *
 * TableHeader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableHeader({ children }) {
  return (
    <div
      className={styles.rowGridContainer}
      style={{ gridTemplateColumns: `repeat(${React.Children.count(children)}, 1fr)` }}
    >
      {children}
    </div>
  );
}

TableHeader.propTypes = {
  children: PropTypes.node,
};

export default TableHeader;
