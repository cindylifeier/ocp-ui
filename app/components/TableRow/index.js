/**
 *
 * TableRow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableRow({ children }) {
  return (
    <div
      className={styles.rowGridContainer}
      style={{ gridTemplateColumns: `repeat(${React.Children.count(children)}, 1fr)` }}
    >
      {children}
    </div>
  );
}

TableRow.propTypes = {
  children: PropTypes.node,
};

export default TableRow;
