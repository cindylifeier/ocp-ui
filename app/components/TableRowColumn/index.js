/**
 *
 * TableRowColumn
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableRowColumn({ children }) {
  return (
    <div className={styles.cellGridItem}>
      {children}
    </div>
  );
}

TableRowColumn.propTypes = {
  children: PropTypes.node,
};

export default TableRowColumn;
