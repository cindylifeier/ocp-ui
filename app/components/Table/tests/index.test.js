import React from 'react';
import { shallow } from 'enzyme';

import Table from '../index';
import TableHeaderColumn from '../../TableHeaderColumn';
import TableHeader from '../../TableHeader';
import TableRow from '../../TableRow';
import TableRowColumn from '../../TableRowColumn';

describe('<Table />', () => {
  it('shoud match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<Table>
        <TableHeader>
          <TableHeaderColumn>Col</TableHeaderColumn>
        </TableHeader>
        <TableRow>
          <TableRowColumn>Row</TableRowColumn>
        </TableRow>
      </Table>),
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
})
;
