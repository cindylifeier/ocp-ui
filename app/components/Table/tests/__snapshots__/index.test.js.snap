// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> shoud match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className={undefined}
    >
        <TableHeader>
            <TableHeaderColumn>
                Col
            </TableHeaderColumn>
        </TableHeader>
        <TableRow>
            <TableRowColumn>
                Row
            </TableRowColumn>
        </TableRow>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className={undefined}
      >
            <TableHeader>
                  <TableHeaderColumn>
                        Col
                  </TableHeaderColumn>
            </TableHeader>
            <TableRow>
                  <TableRowColumn>
                        Row
                  </TableRowColumn>
            </TableRow>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Table>
        <TableHeader>
                <TableHeaderColumn>
                        Col
                </TableHeaderColumn>
        </TableHeader>
        <TableRow>
                <TableRowColumn>
                        Row
                </TableRowColumn>
        </TableRow>
</Table>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <TableHeader>
              <TableHeaderColumn>
                            Col
              </TableHeaderColumn>
</TableHeader>,
            <TableRow>
              <TableRowColumn>
                            Row
              </TableRowColumn>
</TableRow>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className={undefined}
          >
                    <TableHeader>
                              <TableHeaderColumn>
                                        Col
                              </TableHeaderColumn>
                    </TableHeader>
                    <TableRow>
                              <TableRowColumn>
                                        Row
                              </TableRowColumn>
                    </TableRow>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className={undefined}
          >
                    <TableHeader>
                              <TableHeaderColumn>
                                        Col
                              </TableHeaderColumn>
                    </TableHeader>
                    <TableRow>
                              <TableRowColumn>
                                        Row
                              </TableRowColumn>
                    </TableRow>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Table>
    <TableHeader>
        <TableHeaderColumn>
            Col
        </TableHeaderColumn>
    </TableHeader>
    <TableRow>
        <TableRowColumn>
            Row
        </TableRowColumn>
    </TableRow>
</Table>,
}
`;
